jetty-deploy/src/test/java/org/eclipse/jetty/deploy/providers/WebAppProviderTest.java

```diff
@@ -225,12 +229,62 @@ public void testBaseDirSymlink() throws Exception
             server.stop();
         }
     }
-    
-    private Map<String, String> setupJettyProperties(Path jettyHome)
+
+    @Test
+    public void testDelayedDeploy() throws Exception
     {
+        Path realBase = testdir.getEmptyPathDir();
+
+        //set jetty up on the real base
+        jetty = new XmlConfiguredJetty(realBase);
+        jetty.addConfiguration("jetty.xml");
+        jetty.addConfiguration("jetty-http.xml");
+        jetty.addConfiguration("jetty-deploy-wars.xml");
+
+        //Put a webapp into the base
+        jetty.copyWebapp("foo-webapp-1.war", "foo.war");
+
+        Path jettyHome = jetty.getJettyHome().toPath();
+
         Map<String, String> properties = new HashMap<>();
-        properties.put("jetty.home", jettyHome.toFile().getAbsolutePath());
-        return properties;
+        properties.put("jetty.home", jettyHome.toString());
+        properties.put("jetty.deploy.deferInitialScan", "true");
+        //Start jetty, but this time running from the symlinked base
+        System.setProperty("jetty.home", properties.get("jetty.home"));
+
+        List<Resource> configurations = jetty.getConfigurations();
+        Server server = XmlConfiguredJetty.loadConfigurations(configurations, properties);
+
+        try
+        {
+            server.start();
```

Comment:
It would be good if you could get a reference to the `ScanningAppProvider` before the server starts, and add a `LifeCycle.Listener` to it which would assert that `scanningStarted == false`. Otherwise this test isn't really testing that the webapp was not deployed after the `Scanner` started.